### PR TITLE
appimage-run: Add libsecret for bitwarden 

### PR DESCRIPTION
**Pull request by alexyao2015**
## Description of changes

Add  libsecret to appimage-run. The Bitwarden AppImage requires this to function correctly.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/build-support/appimage/default.nix [hunk 1]
```diff
@@ -205,6 +205,8 @@ rec {
       at-spi2-core
       pciutils # for FreeCAD
       pipewire # immersed-vr wayland support
+
+      libsecret
```
[the remark this answers is not in the document]
Comment (alexyao2015) [reply pkgs/build-support/appimage/default.nix]: What's wrong with the commit message and pr title?

**Comment by alexyao2015** [comment]
Done

**Comment by alexyao2015** [comment]
Commit message updated

**Review by philiptaron — APPROVED** [review]

**Comment by philiptaron** [comment]
Result of `nixpkgs-review pr 343845` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>fspy</li>
    <li>golden-cheetah-bin</li>
  </ul>
</details>
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>appimage-run-tests</li>
  </ul>
</details>
<details>
  <summary>128 packages built:</summary>
  <ul>
    <li>altair</li>
    <li>alvr</li>
    <li>anki-bin</li>
    <li>anytype</li>
    <li>appimage-run</li>
    <li>archipelago</li>
    <li>archipelago-minecraft</li>
    <li>arduino-ide</li>
    <li>badlion-client</li>
    <li>bazecor</li>
    <li>beekeeper-studio</li>
    <li>beeper</li>
    <li>betterdiscord-installer</li>
    <li>bloomrpc</li>
    <li>bootstrap-studio</li>
    <li>buttercup-desktop</li>
    <li>caido</li>
    <li>caprine-bin</li>
    <li>chatzone-desktop</li>
    <li>chrysalis</li>
    <li>cider</li>
    <li>clash-nyanpasu</li>
    <li>clickup</li>
    <li>clockify</li>
    <li>code-cursor</li>
    <li>codux</li>
    <li>cozy-drive</li>
    <li>crypto-org-wallet</li>
    <li>davinci-resolve</li>
    <li>davinci-resolve-studio</li>
    <li>dbgate</li>
    <li>deskreen</li>
    <li>devdocs-desktop</li>
    <li>dopamine</li>
    <li>electron-mail</li>
    <li>electronplayer</li>
    <li>everdo</li>
    <li>fflogs</li>
    <li>firefly-desktop</li>
    <li>flexoptix-app</li>
    <li>fluent-reader</li>
    <li>framesh</li>
    <li>gearlever</li>
    <li>hamrs</li>
    <li>hamsket</li>
    <li>heptabase</li>
    <li>hifile</li>
    <li>hoppscotch</li>
    <li>httpie-desktop</li>
    <li>immersed-vr</li>
    <li>insomnia</li>
    <li>irccloud</li>
    <li>jan</li>
    <li>jbrowse</li>
    <li>jetbrains-toolbox</li>
    <li>joplin-desktop</li>
    <li>kchat</li>
    <li>keet</li>
    <li>kmeet</li>
    <li>lbry</li>
    <li>ldtk</li>
    <li>ledger-live-desktop</li>
    <li>lens</li>
    <li>librewolf-bin</li>
    <li>listen1</li>
    <li>lmstudio</li>
    <li>losslesscut-bin</li>
    <li>lunar-client</li>
    <li>lunatask</li>
    <li>marktext</li>
    <li>mathpix-snipping-tool</li>
    <li>mendeley</li>
    <li>miru</li>
    <li>mobilecoin-wallet</li>
    <li>mockoon</li>
    <li>molotov</li>
    <li>motrix</li>
    <li>mqttx</li>
    <li>muffon</li>
    <li>museeks</li>
    <li>mycrypto</li>
    <li>neo4j-desktop</li>
    <li>nosql-workbench</li>
    <li>notable</li>
    <li>notesnook</li>
    <li>notion-app-enhanced</li>
    <li>nuclear</li>
    <li>osu-lazer-bin</li>
    <li>p3x-onenote</li>
    <li>paperlib</li>
    <li>parsify</li>
    <li>pinokio</li>
    <li>plexamp</li>
    <li>polypane</li>
    <li>protonup-qt</li>
    <li>pureref</li>
    <li>quba</li>
    <li>railway-wallet</li>
    <li>rambox</li>
    <li>raven-reader</li>
    <li>remnote</li>
    <li>requestly</li>
    <li>rquickshare</li>
    <li>saleae-logic-2</li>
    <li>session-desktop</li>
    <li>simplex-chat-desktop</li>
    <li>snipaste</li>
    <li>sonixd</li>
    <li>sozi</li>
    <li>ssb-patchwork</li>
    <li>station</li>
    <li>status-im</li>
    <li>steam-rom-manager</li>
    <li>sylk</li>
    <li>timeular</li>
    <li>todoist-electron</li>
    <li>trezor-suite</li>
    <li>tusk</li>
    <li>tutanota-desktop</li>
    <li>upscayl</li>
    <li>via</li>
    <li>vial</li>
    <li>wootility</li>
    <li>wowup-cf</li>
    <li>xlights</li>
    <li>zecwallet-lite</li>
    <li>zettlr</li>
    <li>zulip</li>
  </ul>
</details>

I ran `nix-build -A appimage-run.tests` manually.

**Comment by adminy** [comment]
```bash
evaluating derivation 'path:/etc/nixos/.config/home-manager#homeConfigurations."adminy".activationPacerror: builder for '/nix/store/f429rwc897phs97f27f0w0gkrv74103r-libsecret-0.21.4.drv' failed with exit code 1;
       last 10 log lines:
       > 20/24 libsecret:libsecret / test-collection       FAIL            3.96s   killed by signal 5 SIGTRAP
       >
       > Ok:                 23
       > Expected Fail:      0
       > Fail:               1
       > Unexpected Pass:    0
       > Skipped:            0
       > Timeout:            0
       >
       > Full log written to /build/libsecret-0.21.4/build/meson-logs/testlog.txt
       For full logs, run 'nix log /nix/store/f429rwc897phs97f27f0w0gkrv74103r-libsecret-0.21.4.drv'.
error: 1 dependencies of derivation '/nix/store/prw4lp7q7djig3n8383ynqd4527f94la-appimage-run-usr-multi.drv' failed to build
error: 1 dependencies of derivation '/nix/store/spi5fdbybp9lsbwqql6pfy8wll8jv8md-appimage-run-fhs.drv' failed to build
error: 1 dependencies of derivation '/nix/store/cwy8vd8dm59mk336kzz8bfbnvn35nvj2-appimage-run-init.drv' failed to build
error: 1 dependencies of derivation '/nix/store/9r93ys2rhya1yn6zskl6y6cqcx1vpd6b-appimage-run.drv' failed to build
error: 1 dependencies of derivation '/nix/store/b1hh6z7k6hm0q0cv6x256373dsfpbf39-davinci-resolve-19.0.1.drv' failed to build
error: 1 dependencies of derivation '/nix/store/wf8fka6ff0l0vbk6fardaiyvnj5rhx5h-davinci-resolve-19.0.1.drv' failed to build
error: 1 dependencies of derivation '/nix/store/a1dw2x2dgd9igdsfzv9y7q3m0jks0px9-home-manager-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/qsbgwnqafmnny3lbzlpgq3kvm5i1n5dl-home-manager-generation.drv' failed to build
```

**Pull request merged**